inference-api/inference_api_python_client/client.py

```diff
@@ -0,0 +1,156 @@
+"""Class definition for Inference API Client."""
+
+import os
+
+from typing import Any, Dict, Final, List, Optional
+
+import aiohttp
+
+from asgiref.sync import async_to_sync
+from retry import retry
+
+from .exceptions import (
+  FailedToWriteSegmentationError,
+  InvalidInferenceAPIClientArgsException,
+  SegmentationDownloadError,
+)
+from .django_jwt_client import DjangoJWTClient
+from .types import Inference
+
+class InferenceAPIClient:
+  """Client for interacting with the Bunkerhill Health Inference API.
+  """
+  AUTH_PATH: Final[str] = 'api/auth/jwt_login/'
+  GET_INFERENCE_PATH: Final[str] = 'api/models/{model_id}/patients/{patient_mrn}/inferences/'
+
+  _session: aiohttp.ClientSession
+
+  def __init__(
+    self,
+    username: str,
+    private_key_filename: Optional[str] = None,
+    private_key_string: Optional[str] = None,
+    base_url: str = 'https://api.bunkerhillhealth.com/',
+  ) -> None:
+    """Constructs an InferenceAPIClient.
+
+    Args:
+      username (str): Username for authentication.
+      private_key_filename (str, optional): Path to the private key file for authentication.
+      private_key_string (str, optional): String representation of the private key for authentication.
+      base_url (str, optional): Base URL for the API. Defaults to 'https://api.bunkerhillhealth.com/'.
+    Raises:
+      InvalidInferenceAPIClientArgsException: If invalid arguments are passed to the constructor.
+    """
+    self._validate_args(private_key_filename, private_key_string)
+    if private_key_filename:
+      with open(private_key_filename, 'r') as f:
+        client_private_key = f.read()
+    elif private_key_string:
+      client_private_key = private_key_string
+    self._django_jwt_client = DjangoJWTClient(
+      username=username,
+      client_private_key=client_private_key,
+      base_url=base_url,
+      auth_path=self.AUTH_PATH,
+    )
+
+  async def __aenter__(self):
+    self._session = aiohttp.ClientSession()
+    return self
+
+  async def __aexit__(self, exc_type, exc, tb):
+    await self._session.close()
```

Comment:
I think you'll want to set `self._session = None` after closing the session so it can be garbage collected.